### PR TITLE
Enforce wide-events logging and context conventions with arch tests

### PR DESCRIPTION
**Pull request by fgilio**
## Summary

Implements comprehensive architectural enforcement of the wide-events logging protocol through new CI-checkable rules, adds privacy-safe exception sanitization, and refactors logging infrastructure to ensure consistent, queryable event shapes across success and failure paths.

## Key Changes

**Logging Architecture Tests** (`tests/Arch/LoggingConventionsTest.php`)
- Refactored helper functions to cache file reads and support pattern-based call extraction
- Added `conventionCalls()` for reusable paren-balanced facade call scanning
- Added `contextConventionCalls()` to scan `Context::*` method calls
- Added `staticStringLiteral()` to extract literal string values for static analysis
- New CI-enforced rules:
  - `Context::*` keys must be namespaced under `rfa.`
  - `rfa.outcome` values must use approved vocabulary (`completed`, `error`, `skipped`, `cancelled`, `rejected`, `partial`)
  - Context keys cannot use banned absolute-path names
  - Log and context payloads cannot carry raw exception text (`$e->getMessage()`, `$e->getTraceAsString()`, `$e->stderr`) — must wrap in `LogSanitizer::summary()`

**Exception Sanitization** (`app/Support/LogSanitizer.php`, `tests/Unit/Support/LogSanitizerTest.php`)
- New `LogSanitizer::summary()` utility that scrubs free-form process/exception text:
  - Strips ANSI color sequences
  - Collapses whitespace to single spaces
  - Replaces user's home directory with `~`
  - Truncates overlong text with ellipsis marker
- Comprehensive unit tests covering edge cases (sibling directories, truncation boundaries, etc.)

**Log Channel Posture** (`tests/Feature/LogChannelPostureTest.php`)
- New feature test verifying default log channel resolves to local-only sinks
- Detects remote drivers (`slack`, `syslog`, `errorlog`) and network-bound Monolog handlers
- Recursively expands stack channels to validate all members stay local
- Enforces RFA's no-off-box-logs requirement

**Updater Event Lifecycle** (`app/Providers/NativeAppServiceProvider.php`)
- Refactored `UpdateAvailable`, `UpdateDownloaded`, and `UpdateError` listeners to emit canonical info events on every terminal outcome
- Success paths now emit `Log::info('updater.available')` / `Log::info('updater.downloaded')` with `rfa.outcome = completed`
- Failure path now emits `Log::info('updater.failed')` with `rfa.outcome = error` (not just `Log::error()`)
- All paths record `rfa.duration_ms` via new `elapsedMs()` helper
- Error details wrapped in `LogSanitizer::summary()` to prevent raw exception text leakage
- Context flushed at start, fields added in `finally` block to ensure consistent shape

**Logging Configuration** (`config/logging.php`)
- Changed default channel from `stack` to `daily` (bounded, info-level, 7-day retention)
- Updated single channel default level from `debug` to `info`
- Added environment variable for daily log retention days

**Exception Handling Updates**
- `GitMetadataService`, `GetProjectStatusAction`, `EnsureRfaGitExcludeAction`, `LoadFileDiffAction`, `SyntaxHighlightService`, `OpenProjectFromPathAction`, `ResolveBranchBaseAction`: replaced raw `$e->getMessage()` with `$e->exitCode` or `LogSanitizer::summary()` in warning payloads
- `⚡review-page.blade.php`: wrapped exception text in `LogSanitizer::summary()`

**Documentation** (`.claude/skills/wide-events/SKILL.md`)
- Clarified Context facade usage (thin API slice, deliberate `flush()` requirement for single-process desktop app)
- Expanded canonical event rules: all outcomes must emit info events, listener-owned events measure callback duration unless reliable start time exists
- Added

https://claude.ai/code/session_01QCgoS3hBhpoPi7FCyGkULd

## Summary by CodeRabbit

* **New Features**
  * Improved logging and diagnostics for update, project, and syntax-highlighting flows.
  * Added safer, more consistent log summaries for command failures.

* **Bug Fixes**
  * Replaced raw error text with sanitized details or error class names in logs.
  * Improved update event timing and ensured failure states are still recorded.

* **Tests**
  * Added coverage for logging privacy rules and local-only log channel behavior.
  * Added tests for log sanitization output and truncation behavior.

### CODE REVIEW

File: app/Providers/NativeAppServiceProvider.php [hunk 1]
```diff
@@ -127,41 +134,72 @@ private function registerNativeEventListeners(): void
...
+            $startedAt = microtime(true);
+
+            try {
+                // Diagnostic detail for triage. The canonical info event below
+                // keeps the failure path the same shape as the success paths so
+                // `rfa.outcome` distribution stays queryable across outcomes.
+                Log::error('updater.failed', [
+                    'reason' => 'updater_error',
+                    'message' => LogSanitizer::summary($event->message),
+                    'stack' => LogSanitizer::summary($event->stack, 500),
```
[the remark this answers is not in the document]
Comment (fgilio) [reply app/Providers/NativeAppServiceProvider.php:173]: Good catch — fixed in 76fe0aa. Rather than coalescing at this call site, I made `LogSanitizer::summary()` accept `?string` and treat null as empty: a log sanitizer on an error path shouldn't itself throw, and this covers any nullable field (not just `Error::$stack`) without each caller having to remember `?? ''`. Added a regression test (`summary(null) === ''`).

---
_Generated by [Claude Code](https://claude.ai/code)_
